node/src/components/block_proposer.rs

```diff
@@ -391,21 +409,24 @@ impl BlockProposerReady {
                 break;
             }
 
-            if self.is_deploy_valid(
-                deploy_type.header(),
-                block_timestamp,
-                &deploy_config,
-                &past_deploys,
-            ) && !past_deploys.contains(hash)
+            if self.is_deploy_valid(&deploy_type, block_timestamp, &deploy_config, &past_deploys)
+                && !past_deploys.contains(hash)
                 && !self.sets.finalized_deploys.contains_key(hash)
             {
+                let header = deploy_type.header();
+                let gas_price = header.gas_price();
```

Comment:
Isn't the gas price the number of _tokens per unit of gas_?
What we need to add up here is the "willingness to pay", i.e. the product of the gas price and the deploy's gas limit. (Maybe deploys don't have that yet?)

Reply:
Yes, we should both introduce gas reservations/limits (I prefer "reservation" because CEP#22 makes you pay for everything, not just what you use, refunds aside) if they're not there and check the sum of these against the block gas limit.

Reply:
Per jira comment, we'll be addressing this after CEP22 merges.